Support OTLP exporter for tracing (#13118)
mlflow/mlflow: docs/source/llms/tracing/index.rst

Note that the async logging does not fully eliminate the performance overhead. Some backend calls still need to be made synchronously and there are other factors such as data serialization. However, async logging can significantly reduce the overall overhead of logging traces, empirically about ~80% for typical workloads.

Using OpenTelemetry Collector for Exporting Traces
--------------------------------------------------

> 

Reviewer: Can we make it clear that the configuration of the exporter is exclusive (we can sink traces to only one location) so that users don't wonder why, if they configure an OTEL collector sink, their traces aren't showing up in the MLflow UI?

Author: Makes sense!

> Traces generated by MLflow is compatible with the `OpenTelemetry trace specs <https://opentelemetry.io/docs/specs/otel/trace/api/#span>`_.
> Therefore, MLflow Tracing supports exporting traces to OpenTelemetry Collector, which can then be used to export traces to various backends such as Jaeger, Zipkin, and AWS X-Ray.

Reviewer: ```suggestion
Traces generated by MLflow are compatible with the `OpenTelemetry trace specs <https://opentelemetry.io/docs/specs/otel/trace/api/#span>`_.
Therefore, MLflow Tracing supports exporting traces to an OpenTelemetry Collector, which can then be used to export traces to various backends such as Jaeger, Zipkin, and AWS X-Ray.
```

> By default, MLflow exports traces to the MLflow Tracking Server. To enable exporting traces to OpenTelemetry Collector, set ``OTEL_EXPORTER_OTLP_ENDPOINT`` environment variable (or ``OTEL_EXPORTER_OTLP_TRACES_ENDPOINT``) to the target URL of the OpenTelemetry Collector **before starting any trace**.

Reviewer: ```suggestion
By default, MLflow exports traces to the MLflow Tracking Server. To enable exporting traces to an OpenTelemetry Collector, set the ``OTEL_EXPORTER_OTLP_ENDPOINT`` environment variable (or ``OTEL_EXPORTER_OTLP_TRACES_ENDPOINT``) to the target URL of the OpenTelemetry Collector **before starting any trace**.
```

.. code-block:: python

    import mlflow
    import os

    # Set the endpoint of the OpenTelemetry Collector
    os.environ["OTEL_EXPORTER_OTLP_TRACES_ENDPOINT"] = "http://localhost:4317/v1/traces"
    # Optionally, set the service name to group traces
    os.environ["OTEL_SERVICE_NAME"] = "<your-service-name>"

    # Trace will be exported to the OTel collector at http://localhost:4317/v1/traces
    with mlflow.start_span(name="foo") as span:
        span.set_inputs({"a": 1})
        span.set_outputs({"b": 2})

Configurations
^^^^^^^^^^^^^^

> MLflow uses the standard OTLP Exporter for exporting traces to OpenTelemetry Collector. Thereby, you can use `all the configurations <https://opentelemetry.io/docs/languages/sdk-configuration/otlp-exporter/>`_ supported by OpenTelemetry. The following example configures the OTLP Exporter to use HTTP protocol instead of the default gRPC and sets custom headers:

Reviewer: ```suggestion
MLflow uses the standard OTLP Exporter for exporting traces to OpenTelemetry Collector instances. Thereby, you can use `all of the configurations <https://opentelemetry.io/docs/languages/sdk-configuration/otlp-exporter/>`_ supported by OpenTelemetry. The following example configures the OTLP Exporter to use HTTP protocol instead of the default gRPC and sets custom headers:
```
